overlord/snapshotstate/snapshotstate.go

```diff
@@ -256,6 +256,10 @@ func checkSnapshotTaskConflict(st *state.State, setID uint64, conflictingKinds .
 // Note that the state must be locked by the caller.
 var List = backend.List
 
+// Export a given snapshot ID.
+// XXX: locking? streaming this out is a slow operation
```

Comment:
IMHO we should add the locking to this PR

Reply:
I rewrote the way we export now so that we hopefully do not need locks anymore. Even if a snapshot is removed in the middle of the export we can still finish the export.